src/components/app/data/utils.js

```diff
@@ -470,28 +472,45 @@ export function getAvailableCourseRuns({ course, isEnrollableBufferDays }) {
   if (!course?.courseRuns) {
     return [];
   }
-  const availableCourseRunsFilter = (courseRun) => {
-    if (!courseRun.isMarketable || isArchived(courseRun)) {
-      return false;
-    }
 
-    if (isEnrollableBufferDays === undefined) {
-      return courseRun.isEnrollable;
+  // These are the standard rules used for determining whether a run is "available".
+  const standardAvailableCourseRunsFilter = (courseRun) => (
+    courseRun.isMarketable && !isArchived(courseRun) && courseRun.isEnrollable
+  );
+
+  // These are more relaxed availability rules when late enrollment is applicable. We still never show archived courses,
+  // but the rules around the following fields are relaxed:
+  //
+  // * courseRun.isEnrollable: This field represents the enrollment window actually stored in the database. However,
+  //   during late enrollment we expand the end date of the enrollment window by isEnrollableBufferDays.
+  // * courseRun.isMarketable: This field is True when the run is published, has seats, and has a marketing URL. Since
+  //   late enrollment potentially means enrolling into an unpublished run, we must ignore the run state.
+  const lateEnrollmentAvailableCourseRunsFilter = (courseRun) => {
+    // If any of these early checks are True, then fail fast.
+    const baseNonAvailableChecks = [
+      isArchived(courseRun),
+      // The next two checks are in lieu of isMarketable which is otherwise overly sensitive to courserun state.
+      !('seats' in courseRun) || !courseRun.seats?.length,
+      !('marketingUrl' in courseRun) || !courseRun.marketingUrl,
+    ];
+    if (baseNonAvailableChecks.some(Boolean)) {
+      return false;
     }
 
+    // Finally, check against an expanded enrollment window.
     const today = dayjs();
-    if (courseRun.enrollmentStart && today.isBefore(dayjs(courseRun.enrollmentStart))) {
-      // In cases where we don't expect the buffer to change behavior, fallback to the backend-provided value.
-      return courseRun.isEnrollable;
-    }
-    if (!courseRun.enrollmentEnd) {
+    if (!courseRun.enrollmentEnd || (courseRun.enrollmentStart && today.isBefore(dayjs(courseRun.enrollmentStart)))) {
       // In cases where we don't expect the buffer to change behavior, fallback to the backend-provided value.
-      return courseRun.isEnrollable;
+      return standardAvailableCourseRunsFilter(courseRun);
     }
     const bufferedEnrollDeadline = dayjs(courseRun.enrollmentEnd).add(isEnrollableBufferDays, 'day');
     return today.isBefore(bufferedEnrollDeadline);
   };
-  return course.courseRuns.filter(availableCourseRunsFilter);
+
+  // isEnrollableBufferDays is used as a heuristic to determine if the late enrollment feature is enabled.
+  return course.courseRuns.filter(
+    isEnrollableBufferDays === undefined ? standardAvailableCourseRunsFilter : lateEnrollmentAvailableCourseRunsFilter,
```

Comment:
[nit/question] is there any chance of `isEnrollableBufferDays` to return as a null? Could `!isEnrollableBufferDays` achieve the same results?

Reply:
I was thinking along the same lines, @brobro10000. But `!isEnrollableBufferDays` could, in theory, be `0` would be truthy with the check on `!isEnrollableBufferDays` (when it is defined, it returns the value of a constant `30` but what if it were to change to `0` for some reason, or otherwise be configurable/dynamic).

Maybe checking for whether the value is a number? The (below) latter `!isNaN` check is because `typeof NaN === 'number'` is also true.

```jsx
typeof isEnrollableBufferDays === 'number' && !isNaN(isEnrollableBufferDays) ? lateEnrollmentAvailableCourseRunsFilter : standardAvailableCourseRunsFilter,
```

Another alternative suggestion might be use the util functions in this repo like `isDefined`, `isDefinedAndNotNull`, etc.

```jsx
isDefinedAndNotNull(isEnrollableBufferDays) ? lateEnrollmentAvailableCourseRunsFilter : standardAvailableCourseRunsFilter,
```

[unrelated nit / follow-up] I wonder if we might want to consider eventually renaming `isEnrollableBufferDays` since it seems to indicate its value is a boolean due to the `is` prefix, but it returns an integer constant or isn't defined. Perhaps something like `lateEnrollmentBufferDays` would make sense.